packages/hub/src/lib/commit.ts

```diff
@@ -459,6 +467,38 @@ export async function* commitIter(params: CommitParams): AsyncGenerator<CommitPr
 
 		yield { event: "phase", phase: "committing" };
 
+		const allOpsToCommit = [...deleteOps, ...fileOpsToCommit];
+		if (allOpsToCommit.length === 0) {
+			if (!currentCommitOid) {
```

Comment:
IIUC, we currently don't throw an error in the case where `operations` is empty, meaning it's possible to get there without calling the `preupload` endpoint before, and thus `currentCommitOid` can still be `undefined` here.

So I kept this for now; the other alternative would be to throw an error if there are no operations, but it would be breaking change - I'll let you decide @coyotte508

Reply:
It's fine to throw on empty commits IMO, we can do a major version change (and maybe do some of https://github.com/huggingface/huggingface.js/milestone/2)

Reply:
actually if we upgrade to 1.0 we can add the `null` return type and return `null` on empty commit

(And it's better than throwing an error, at least it will be caught by TS)